spec/PeriodicBackgroundSync-index.bs

```diff
@@ -157,3 +159,51 @@ The <code><dfn method for=PeriodicSyncManager title="getTags()">getTags()</dfn><
 1. Let <var>currentTags</var> be a new [=sequence=].
 1. For each <var>registration</var> in |serviceWorkerRegistration|'s [=list of periodicsync registrations=], add |registration|'s [=tag=] to |currentTags|.
 1. [=Resolve=] |promise| with |currentTags|.
+
+## The <dfn>periodicSync event</dfn> ## {#periodicSync-event}
+<script type="idl">
+dictionary PeriodicSyncEventInit : ExtendableEventInit {
+    required DOMString tag;
+};
+
+[
+    Constructor(DOMString type, PeriodicSyncEventInit init),
+    Exposed=ServiceWorker
+] interface PeriodicSyncEvent : ExtendableEvent {
+    readonly attribute DOMString tag;
+  };
+</script>
+
+The {{PeriodicSyncEvent}} interface represents a [=firing=] [=periodicsync registration=].
+
+### Firing periodicSync events ### {#firing-periodicsync-events}
+Whenever the user agent changes to [=online=], it SHOULD [=fire a periodicSync event=] for each [=periodicsync registration=] whose [=registration state=] is [=pending=] and [=time to fire=] is now or in the past. The events MAY be fired in any order.
+
+### [=Fire a periodicSync event=] ### {#firing-a-periodicsync-event}
+The user agent will fire a [=periodicSync event=] for a [=periodicsync registration=] as soon as network connectivity is available, at or after the [=time to fire=] of the [=periodicsync registration=].
+If a [=periodicSync event=] fails, the user agent MAY <dfn>retry</dfn> it one or more times at a time of its choosing, based on some user agent defined heuristics. The interval between [=retry=] attempts SHOULD be much smaller than the [=minInterval=] value associated with the [=periodicsync registration=].
+
+A user agent MAY impose a time limit on the lifetime extension and execution time of a {{PeriodicSyncEvent}} which is stricter than the time limit imposed for {{ExtendableEvent}}s in general. In particular, any retries of the {{PeriodicSyncEvent}} MAY have a significantly shortened time limit.
+
+To <dfn>fire a periodicSync event</dfn> for a [=periodicsync registration=] <var>registration</var>, the user agent MUST run the following steps:
+1. [=Assert=]: |registration|'s [=registration state=] is [=pending=].
+1. Let <var>serviceWorkerRegistration</var> be the [=service worker registration=] associated with |registration|.
+1. [=Assert=]: |registration| exists in the [=list of periodicSync registrations=] associated with |serviceWorkerRegistration|.
```

Comment:
I've got mixed feelings about this assertion. It implies there's a possibility that `registration` might be orphaned, or be reparented onto another SWR. Neither is true, which makes the assert redundant.

Reply:
Fair. I'll remove this. There's an equivalent assert in BackgroundSync spec, which I'll send a PR to update.